proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -5017,6 +5017,13 @@ lemma sts_st_tcb_at'_cases:
   apply fastforce
   done
 
+lemma sts_st_tcb_at'_cases_strong:
+  "\<lbrace>\<lambda>s. tcb_at' t s \<longrightarrow> (t = t' \<longrightarrow> P (P' ts)) \<and> (t \<noteq> t' \<longrightarrow> P (st_tcb_at' P' t' s))\<rbrace>
+     setThreadState ts t
```

Comment:
Indentation.

Reply:
I would also use `if then else` but that's just a matter of stlye.

Reply:
I think the rough consensus is to use `(P -> X) /\ (!P - Y)` for syntactically small `P`, since it means you don't need to add another lifting lemma to your `wp` proofs.

Reply:
In general, the implication form is nicer than `if` because Isabelle likes to split `if`s by default and pull out the implication to the largest scope, which may be outside the hoare triple (that's why we see a lot of `split del: if_split` in `wp` proofs). For this one here that is likely to happen, because both `t` and `t'` are in scope outside the precondition. 

Only when `P` becomes so large that it's annoying to repeat I'd opt for `if`.